protocol/messenger_communities.go

```diff
@@ -3536,3 +3539,52 @@ func (m *Messenger) AddCommunityToken(token *communities.CommunityToken) (*commu
 func (m *Messenger) UpdateCommunityTokenState(contractAddress string, deployState communities.DeployState) error {
 	return m.communitiesManager.UpdateCommunityTokenState(contractAddress, deployState)
 }
+
+// UpdateCommunityEncryption takes a community and encrypts / decrypts the community
+// based on TokenPermission. Community is republished along with any keys if needed.
+//
+// Note: This function cannot decrypt previously encrypted messages, and it cannot encrypt previous unencrypted messages.
+// This functionality introduces some race conditions:
+//   - community description is processed by members before the receiving the key exchange messages
+//   - members maybe sending encrypted messages after the community description is updated and a new member joins
+func (m *Messenger) UpdateCommunityEncryption(community *communities.Community) (*MessengerResponse, error) {
+	if community == nil {
+		return nil, errors.New("community is nil")
+	}
+
+	becomeMemberPermissions := community.TokenPermissionsByType(protobuf.CommunityTokenPermission_BECOME_MEMBER)
+	isEncrypted := len(becomeMemberPermissions) > 0
+
+	// Check isEncrypted is different to Community's value
+	// If not different return
+	if community.Encrypted() == isEncrypted {
+		return nil, nil
+	}
+
+	if isEncrypted {
+		// 🪄 The magic that encrypts a community
+		_, err := m.encryptor.GenerateHashRatchetKey(community.ID())
+		if err != nil {
+			return nil, err
+		}
+
+		err = m.SendKeyExchangeMessage(community.ID(), community.GetMemberPubkeys(), common.KeyExMsgReuse)
+		if err != nil {
+			return nil, err
+		}
+	}
+
+	// 🧙 There is no magic that decrypts a community, we just need to tell everyone to not use encryption
```

Comment:
This comment is a little bit confusing (however this is mainly due to my lack of knowledge).
When and how do we "tell everyone to not use encryption" ?

I believe nodes check on `community.Encrypted()` themselves and then act accordingly no?

Reply:
From what I know - yes, they receive community updates with the community.Encrypted() == false and act accordingly.
Again, I just refactored the implementation not to use API, but to set encryption automatically.